x/fswap/types/msgs.go

```diff
@@ -141,3 +144,20 @@ func (m *MsgSetSwap) Route() string {
 func (m *MsgSetSwap) GetSignBytes() []byte {
 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(m))
 }
+
+func (m *QuerySwapRequest) Validate() error {
+	if m == nil {
+		return status.Error(codes.InvalidArgument, codes.InvalidArgument.String())
+	}
+	if m.GetFromDenom() == m.GetToDenom() {
+		return status.Error(codes.InvalidArgument, codes.InvalidArgument.String())
```

Comment:
It would be nice to provide the specific reason.

Reply:
I added descriptive messages for invalid `QuerySwapRequest`